modules/programs/zsh.nix

```diff
@@ -109,15 +200,52 @@ in
 
         HELPDIR="${pkgs.zsh}/share/zsh/$ZSH_VERSION/help"
 
-        ${if cfg.enableCompletion then "autoload -U compinit && compinit" else ""}
+        ${concatStrings (map (plugin: ''
+          path+="$HOME/.zsh/plugins/${plugin.name}"
+          fpath+="$HOME/.zsh/plugins/${plugin.name}"
+        '') cfg.plugins)}
+
+        ${optionalString cfg.enableCompletion "autoload -U compinit && compinit"}
         ${optionalString (cfg.enableAutosuggestions)
           "source ${pkgs.zsh-autosuggestions}/share/zsh-autosuggestions/zsh-autosuggestions.zsh"
         }
 
+        ${optionalString cfg.oh-my-zsh.enable ''
+            # oh-my-zsh configuration generated by NixOS
+            export ZSH=${pkgs.oh-my-zsh}/share/oh-my-zsh
+            export ZSH_CACHE_DIR=''${XDG_CACHE_HOME:-$HOME/.cache}/oh-my-zsh
+
+            ${optionalString (cfg.oh-my-zsh.plugins != [])
+              "plugins=(${concatStringsSep " " cfg.oh-my-zsh.plugins})"
+            }
+            ${optionalString (cfg.oh-my-zsh.custom != "")
+              "ZSH_CUSTOM=\"${cfg.oh-my-zsh.custom}\""
+            }
+            ${optionalString (cfg.oh-my-zsh.theme != "")
+              "ZSH_THEME=\"${cfg.oh-my-zsh.theme}\""
+            }
+            source $ZSH/oh-my-zsh.sh
+        ''}
+
+        ${concatStrings (map (plugin: ''
+          source "$HOME/.zsh/plugins/${plugin.name}/${plugin.file}"
+        '') cfg.plugins)}
+
         ${aliasesStr}
 
         ${cfg.initExtra}
       '';
-    }
-  );
+    })
+    (mkIf cfg.oh-my-zsh.enable {
+      programs.zsh.enableCompletion = mkForce false;
+    })
+    (mkIf (builtins.length cfg.plugins > 0) {
+      programs.zsh.enableCompletion = mkDefault true;
+
+      home.file = builtins.listToAttrs (map (plugin: {
```

Comment:
You can avoid `listToAttrs` here since `home.file` accepts lists of sets (the path is in the `target` attribute). Something like
```
home.file = map (plugin: {
  target = ".zsh/plugins/${plugin.name}";
  source = plugin.src;
}) cfg.plugins;
```
should work.